File: src/BooksReader.Web/ClientApp/src/app/modules/core/models/api/entities/public-page.ts
import { IIdentifiable } from './identifiable';
import { PublicPageType } from '@br/core/enums';

export interface PublicPage extends IIdentifiable {
    id: string;

    pageType: PublicPageType;
    subjectId?: string;
    domainId: string;
    seoInfoId?: string;

    urlPath: string;
    content: string;
}

File: src/BooksReader.Core/Entities/PublicPage.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text;
using BooksReader.Core.Enums;
using BooksReader.Core.Models;

namespace BooksReader.Core.Entities
{
    public class PublicPage: IIdentified
    {
        public Guid Id { get; set; }

        public PublicPageType PageType { get; set; }

        public Guid? SubjectId { get; set; }

        public Guid DomainId { get; set; }

        [MaxLength(256)]
        public string UrlPath { get; set; }

        public string Content { get; set; }

        public Guid? SeoInfoId { get; set; }


        /* Navigation properties */
        [ForeignKey("SeoInfoId")]
        public SeoInfo SeoInfo{ get; set; }

        [ForeignKey("DomainId")]
        public UserDomain Domain { get; set; }

    }
}
